sql/plan/flush.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/grant_tables"
+)
+
+// Flush handles all flush statements. https://dev.mysql.com/doc/refman/8.0/en/flush.html
+type Flush struct {
+	option    *sqlparser.FlushOption
+	flushType string
+	gt        grant_tables.GrantTables
```

Comment:
This isn't getting set anywhere, and since this isn't a pointer this is creating a new `GrantTables` struct that will never have a persistence function set on it. That's why it didn't error in the test.

Also since there's no test that's testing whether the statement works, it wasn't caught that we're never calling the actual persist function here.

Look into the `ShowGrants` node in `sql/plan/show_grants.go` and see how it's handling the reference to `GrantTables`.